intercept/chatcompletions/base.go

```diff
@@ -0,0 +1,249 @@
+package chatcompletions
```

Comment:
I don't really understand the package structure of `intercept`.

The high-level packages (`chatcompletions`, `messages`, `eventstream`) are all for different purposes or providers. This also means that code-reuse across different endpoint handlers for the same provider can't share code within their same package, so you'd have to export those - which would be awkward.

Reply:
My idea behind `intercept` package is for it to contain all interceptor implementations + common logic between interceptors. Each sub packages is specific interceptor implementation, each sub package implementing interceptor interface. It may require different structure later as this is the place where most code will probably be added, I can see it becoming too large / new "root" package.

`eventstream` is used in both `chatcompletions` and `messages` packages. `envetstream` isn't useful outside of interception context so I located in `intercept` package as common logic between interceptors. I don't see it being in separate sub-package as awkward.